### DISCUSSION TITLE
Priorisierung der Bedarfe für Patient merge

### DISCUSSION THREAD

**Discussion by f-peverali** [gematik/Spec-ISiK-Basismodul#320] [Polls]
Für die Identifikation und Priorisierung von Bedarfen rund um das Thema 'Patient merge' wollen wir gerne die Meinungen möglichst vieler Stakeholder aus der Community einholen. Daher hier eine Abstimmung zur Erhebung eines Meinungsbilds.

Die unten gelisteten Optionen spiegeln die Bedarfe im [Konzept Draft](https://github.com/gematik/spec-ISiK-Basismodul/blob/Patient-merge/ImplementationGuide/markdown/UebergreifendeFestlegungen/Patient_merge.md) wieder.

Falls weitere Bedarfe bestehen, bitte entsprechend direkt hier als Kommentar einbringen.

**Comment by patrick-werner** [comment]
Wie gehen wir damit um wenn der inform nicht alle relevanten Systeme erreicht (bspw. Netzwerk down)?
Vorschlag:
Zusätzlich zum inform werden patient.link beider Patientobjekte gesetzt, und der nicht mehr gültige Patient auf active = false gesetzt

**Reply by KaiMindermann** [reply]
Subscriptions können best-effort (unreliable) oder guaranteed (reliable) Notifications senden. Im Falle von best-effort sind die Events fortlaufend durchnummeriert und der Client muss erkennen, dass hier Lücken aufgetreten sind. Über die $events-Operation können fehlende Events nachgelesen werden (die Events müssen also auf dem Server zumindest vorübergehend persisitert werden). Im Falle von guaranteed Notifications erkennt der Server, dass kein positives Ergebnis vom Empfänger der Notification kommt (timeout oder negatives OperationOutcome) und dann muss der Server das wiederholen (parametriesiert über die Subscription). Wenn die Fehlerschwellwerte erreicht sind, geht die Subscription in den error-Status. 
https://hl7.org/fhir/subscriptions.html#guaranteed-vs-best-effort 

patient.link wird aus meiner Sicht immer benötigt, denn irgendwie muss ich ja mitteilen, welcher neu bzw. alt gültig ist / war (auch im Notification-Bundle). Bei Systemen ohne Subscription (nur READ-Interation) wird dies auch immer benötigt werden. 

Nächste Option wäre eine Provenance-Instanz mit der die Zusammenführung protokolliert wird und abrufbar wird. Diese Ressource würde eine Alternative darstellen, wenn das KIS den alten Patienten sofort aus der aktiven Datenbank eliminiert. das würde ich aber rein optional ansehen und nachrangig priorisieren. 

Eine Ergänzung noch zu guaranteed vs. best-effort: Da bei best-effort eine lfd. Event-Nummer erforderlich ist, ist das für mich in Systemen mit high-volume Verarbeitung wie in einem KIS ein absolutes No-Go. Alle Parallelisierung wäre dahin, weil nur ein einzelner Prozess die Event-Nummern vergeben kann, damit keine Lücken entstehen oder Überholer auftreten.

**Reply by patrick-werner** [reply]
Vielen Dank für ihren Kommentar und den Hinweis auf best-effort und guaranteed Notifications/Subscriptions. Da dies ein FHIR R5 Feature ist können wir dieses aktuell in ISiK nicht verwenden.

**Reply by KaiMindermann** [reply]
Sorry, wenn ich das so sage, aber nur mit best-effort kann ich in der Praxis keinen Blumentopf gewinnen, wenn ich Systeme miteinander kommunizieren will. Für reine Clients mit sessionbasierten Subscriptions funktioniert das, aber es funktioniert definitiv nicht, wenn ich über Monate und Jahre Datenbestände synchronisieren möchte. Könnt Ihr da bitte nachhaken, ob das nicht Bestandteil des Downports sein kann bzw. ob mit einem eigenen channel-type in der extension http://hl7.org/fhir/uv/subscriptions-backport/StructureDefinition/backport-channel-type eine guaranteed-Notification abbildbar wäre?

**Reply by markus020** [reply]
Hier müsste man den (Business) Impact eines manuellen Nachpflegens in einem Subsystem vs. dem Impact auf ein KIS/KAS betrachten. Heute wird größtenteils mittels HL7 V2 ADT über einen Merge im KIS informiert und "merge" Patient ins Archiv verschoben. Mit Patient.link verpflichtet man die KIS einer falschen Doublette eines Patienten in seinem Datenbestand umzugehen. Dabei ist zu berücksichtigen, dass es auch abseits von Merge Verwendung für Patient.active geben kann (z.B. false = Patient nicht in Klinik in Behandlung) etc. 

Hier kann Subscription zwar eine Alternative sein aber auch hier müssen einige Punkte beachtet werden. V.a. macht FHIR hier keine Vorgaben zum Übertragungsweg, weshalb man hier einerseits Gefahr läuft, dass Subsysteme x-verschiedene Standards umsetzen oder man durch Festlegungen den Aufwand in den KIS und somit auch die Kosten steigert. Hier wäre nochmal mein Vote für eine (ggf. Release-Übergreifende "Infrastruktur-AG", da hier außerhalb des fachlichen sehr viel technisches zu berücksichtigen ist).

Provenance kann hier eine Möglichkeit sein. Insbesondere da wir hier eine zusätzliche Datenstruktur haben, was i.d.R. eine Integration einfacher macht als Änderungen am Bestandssystem.

**Reply by KaiMindermann** [reply]
@patrick-werner : habe noch ein wenig geforscht und hier einen Hinweis im Backport-IG gefunden: https://build.fhir.org/ig/HL7/fhir-subscription-backport-ig/channels.html#custom-channels Es wäre also mit custom-channels auch in R4 mit dem Backport eine guaranteed delivery aufsetzbar..

**Reply by KaiMindermann** [reply]
@markus020 : würdest Du es denn für vertretbar halten, bei einem Request zu einer Patientenressource, die gemerged wurde und bei Euch nicht mehr in der aktuellen DB vorhanden ist, vor dem Liefern der 404-Response in die Provenance reinzuschauen und dann eine synthetisierte Patientenressource mit dem patient.link auf den neuen Patienten zu liefern? Wäre sicherlich keine perfekte Lösung, aber vielleicht ein inhaltlich und wirtschaftlich vertretbarer Kompromiss.

**Reply by patrick-werner** [reply]
Vielen Dank für ihren Input @KaiMindermann  
Ich verstehe den Wunsch nach guaranteed delivery Optionen, aber selbst in R5 wird sich so darauf bezogen:
> Unless otherwise specified by a server implementation and channel, the Subscriptions Framework does not involve [guaranteed ](https://www.rfc-editor.org/rfc/rfc2212)icon delivery of notifications; channels are assumed to be [simple best-effort ](https://www.rfc-editor.org/rfc/rfc5290)icon unless otherwise specified. While the Subscriptions Framework is able to support such mechanisms, defining channels using them is beyond the scope of the standard at this time.
https://hl7.org/fhir/subscriptions.html#overview-notification

Custom Channel definitionen könnte man nutzen um guaranteed delivery channels erzeugen zu können. Nur sind wir hier dann außerhalb der Definition des backport IGs oder des R5 Standards, also weniger interoperabel. Und außerhalb heutiger erhältlicher Softwarelösungen. 
Ein sehr spannendes Thema, ich freue mich auf den Austausch im nächsten AK Meeting und weitere Meinungen hier im Issue.

**Reply by markus020** [reply]
@KaiMindermann: Aus KIS-Sicht wäre das eher suboptimal aus insbesondere folgenden Gründen: 

1) i.d.R. gibt es für den Zugriff auf archivierte Daten gesonderte und stark eingeschränkte Regelungen. Diese würden damit faktisch außer Kraft gesetzt, indem quasi eine Hintertüre geschaffen wurde. 
2) das KIS müsste hier prüfen, weshalb die Ressource nicht mehr vorhanden ist (Möglichkeit wäre u.a. Umsetzung von DSGVO-Forderungen) und dann anhand einer nur hierfür umzusetzenden Berechtigungslogik die Ressource freigeben
3) das (ohnehin schon sehr komplexe) Berechtigungsmanagement müsste erweitert werden. Hier ist die Frage, ob das Subsystem anhand der Berechtigungen vor dem Löschen ein Zugriff gewährt werden kann oder, ob hier anderweitige Rechte eingeführt werden müssen. 

Insgesamt würde ich das nicht als praktikable Lösung sehen.

**Reply by KaiMindermann** [reply]
@markus020 : vielen dank für die detailreiche Information. Dass das aus Entwicklungssicht nicht optimal ist, ist auf jeden Fall klar. Die Kernfrage aus meiner Sicht ist, wie hoch man den Grad der Interoperabilität - also der nahtlosen Zusammenarbeit der Systeme - ansetzt und fehlende Interoperabilität (also Nahtstellen, die vom Anwender kompensiert werden müssen) auf Akzeptanz stößt und damit zu welchen Kompromissen alle Parteien bereit sind.

**Reply by KaiMindermann** [reply]
@patrick-werner : ich habe die Frage nach guaranteed delivery channel gestern Abend in einer Q&A an Gino Canessi richten können. Aus seiner Sicht sind guaranteed delivery channels Bestandteil des backport IGs und eben eine der beiden Tode, die man sterben kann... Entweder best-effort mit serialisierter Eventid und der Client muss fehlende Events detektieren und nachholen oder guaranteed notification und dann eben mit etwas mehr Aufwand in der Pub/Sub Verwaltung beim Server.
